ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```
@@ -0,0 +1,115 @@
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.ReplChangeManager;
+import org.apache.hadoop.hive.ql.exec.ReplCopyTask;
+import org.apache.hadoop.hive.ql.parse.EximUtil;
+import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+public class FileOperations {
+  private static Logger logger = LoggerFactory.getLogger(FileOperations.class);
+  private final Path dataFileListPath;
+  private final Path exportRootDataDir;
+  private HiveConf hiveConf;
+  private final FileSystem dataFileSystem, exportFileSystem;
+
+  public FileOperations(Path dataFileListPath, Path exportRootDataDir, HiveConf hiveConf)
+      throws IOException {
+    this.dataFileListPath = dataFileListPath;
+    this.exportRootDataDir = exportRootDataDir;
+    this.hiveConf = hiveConf;
+    dataFileSystem = dataFileListPath.getFileSystem(hiveConf);
+    exportFileSystem = exportRootDataDir.getFileSystem(hiveConf);
+  }
+
+  public void export(ReplicationSpec forReplicationSpec) throws IOException, SemanticException {
+    if (forReplicationSpec.isLazy()) {
+      exportFilesAsList();
+    } else {
+      copyFiles();
+    }
+  }
+
+  /**
+   * This writes the actual data in the exportRootDataDir from the source.
+   */
+  private void copyFiles() throws IOException {
+    RemoteIterator<LocatedFileStatus> itr = dataFileSystem.listFiles(dataFileListPath, true);
+    while (itr.hasNext()) {
+      LocatedFileStatus fileStatus = itr.next();
+      if (shouldExport(fileStatus)) {
+        ReplCopyTask.doCopy(exportRootDataDir, exportFileSystem, fileStatus.getPath(), dataFileSystem, hiveConf);
```

yeh from a use case perspective that can happen during export, though it might be better to fail there rather than silently move on as there is concurrent write operation happening while export, if not please create a bug for the same.